sfputil/main.py

```diff
@@ -646,13 +683,19 @@ def presence(port):
         for physical_port in physical_port_list:
             port_name = get_physical_port_name(logical_port_name, i, ganged)
 
-            try:
-                presence = platform_chassis.get_sfp(physical_port).get_presence()
-            except NotImplementedError:
-                click.echo("This functionality is currently not implemented for this platform")
-                sys.exit(ERROR_NOT_IMPLEMENTED)
+            if is_rj45_port_from_db(port_name, state_db):
+                status = state_db.get_all(state_db.STATE_DB, 'TRANSCEIVER_STATUS|{}'.format(port_name))
+                status_string = status.get('error')
+                if status_string == 'N/A':
```

Comment:
Dont understand this logic. Can you please elaborate?

Reply:
`N/A` means no error for a port in `TRANSCEIVER_STATUS|<port>.error`.
As [discussed](https://github.com/keboliu/sonic-buildimage/pull/41#discussion_r810984700), if there is no error, the port is present.